### DISCUSSION TITLE
JReleaser error without details

### DISCUSSION THREAD

**Discussion by gotson** [jreleaser/jreleaser#1233] [Q&A]
Hello,

my JReleaser release on CI failed with no details whatsoever: https://github.com/gotson/komga/actions/runs/4322026365/jobs/7544068802#step:11:838

```
INFO]  JReleaser 1.5.0
[INFO]    - basedir set to /home/runner/work/komga/komga/komga
[INFO]    - outputdir set to /home/runner/work/komga/komga/komga/build/jreleaser
[INFO]  git-root-search set to true
[INFO]  Loading variables from /home/runner/.jreleaser/config.properties
[WARN]  Variables source /home/runner/.jreleaser/config.properties does not exist
[INFO]  Validating configuration
[INFO]  Strict mode set to false
[INFO]    [validation] Adding registry docker.io to distribution.komga.docker with username '***'
[INFO]  Project version set to 0.162.0
[INFO]  Release is not snapshot
[INFO]  Timestamp is 2023-03-03T09:31:51.906Z
[INFO]  HEAD is at cb332a6
[INFO]  Platform is linux-x86_64
[INFO]  dry-run set to false
[INFO]  Generating changelog
[INFO]  Extracting issues from changelog
[INFO]  Storing issues: build/jreleaser/release/issues.txt
[INFO]  Storing changelog: build/jreleaser/release/CHANGELOG.md
[INFO]  Storing changelog: ../CHANGELOG.md
[INFO]  Cataloging artifacts
[INFO]    [sbom] Cataloging is not enabled. Skipping
[INFO]  Calculating checksums for distributions and files
[INFO]    [checksum] build/libs/komga-0.162.0.jar.sha256
[INFO]  Signing distributions and files
[INFO]    [sign] Signing is not enabled. Skipping
[INFO]  Deploying Maven artifacts
[INFO]    [maven] Deploying is not enabled. Skipping
[INFO]  Uploading distributions and files
[INFO]    [upload] Uploading is not enabled. Skipping
[INFO]  Releasing to https://github.com/***/komga
[INFO]   - uploading checksums_sha256.txt
[INFO]   - uploading komga-0.162.0.jar
[INFO]   - uploading komga-0.162.0.jar.sha256
[WARN]  Could not update release v0.162.0 with discussion category Announcements
[INFO]  Marking 2 issue(s) as released
[INFO]  Preparing distributions
[INFO]    - Preparing komga distribution
[INFO]      [docker] preparing komga distribution
[INFO]  Packaging distributions
[INFO]    - Packaging komga distribution
[INFO]      [docker] packaging komga distribution
[INFO]      [docker]  - docker.io/***/komga:latest
[INFO]      [docker]  - docker.io/***/komga:0.162.0
[INFO]      [docker]  - docker.io/***/komga:0.x
[INFO]      [docker]  - ghcr.io/***/komga:latest
[INFO]      [docker]  - ghcr.io/***/komga:0.162.0
[INFO]      [docker]  - ghcr.io/***/komga:0.x
[INFO]  Publishing distributions
[INFO]    - Publishing komga distribution
[INFO]      [docker] publishing komga distribution
[INFO]      [docker] setting up repository ***/komga-docker
[INFO]    Announcing release
[INFO]      [announce] Announcing is not enabled. Skipping
[INFO]  Writing output properties to build/jreleaser/output.properties
[ERROR] JReleaser failed after 01:00 m
```

Any idea what happened ?

The release was created on Github, the files were attached.

However the Docker images were not pushed to the registries, and the closed issues were not commented upon,

**Comment by aalmiray** [comment]
Were you able to capture the `trace.log` file?

**Reply by gotson** [reply]
it was run on CI, so unfortunately no

**Reply by aalmiray** [reply]
Unfortunate, yes. Will try to replicate locally. May I suggest to upload the logs when running on CI? Here’s how JReleaser does it https://github.com/jreleaser/jreleaser/blob/36b04bf3818c6233288b29626601f323ed0ae461/.github/workflows/early-access.yml#L114 You’ll have to adjust the paths as I believe your project uses the Gradle plugin

**Reply by gotson** [reply]
When running locally it also crashes, here is the trace:
[trace.log](https://github.com/jreleaser/jreleaser/files/10880768/trace.log)

**Reply by gotson** [reply]
`Caused by: org.jreleaser.model.spi.packagers.PackagerProcessingException: Unexpected error updating gotson/komga-docker`

I have no idea what `gotson/komga-docker` is though

**Reply by gotson** [reply]
My full JReleaser config is here: https://github.com/gotson/komga/blob/master/komga/build.gradle.kts#L310-L417

**Reply by aalmiray** [reply]
> I have no idea what `gotson/komga-docker` is though

I believe this is a remote Git repository where the generated Dockerfile and artifacts get pushed to keep records. You can disable this by setting 

```
packagers:
  docker:
    repository:
      active: NEVER
```

**Reply by gotson** [reply]
I already have this set: https://github.com/gotson/komga/blob/b2d3b3b3372c0bc676f54f601be095d4b740b01d/komga/build.gradle.kts#L391

**Reply by aalmiray** [reply]
I see it now. This seems like a bug as the repository was still enabled judging by the resolved configuration 

```
 repository:
                enabled: true
                active: release
                owner: gotson
                name: komga-docker
                tagName: v{{projectVersion}}
                branch: HEAD
                username: gotson
                token: ************
                commitMessage: {{distributionName}} {{tagName}}
                versionedSubfolders: false
```

This bug may have been introduced in `1.5.0` when resolving the value of `active` via env vars.

**Reply by gotson** [reply]
even after setting `JRELEASER_DISTRIBUTIONS_komga_DOCKER_REPOSITORY_ACTIVE` to `false`, `jreleaserConfig` will still consider it `true`

**Comment by aalmiray** [comment]
komga must be in uppercase as well

**Reply by gotson** [reply]
That works, when setting it to `NEVER`

**Reply by gotson** [reply]
FYI i tried setting `JRELEASER_DOCKER_REPOSITORY_ACTIVE` only, but that doesn't work. `JRELEASER_DISTRIBUTIONS_KOMGA_DOCKER_REPOSITORY_ACTIVE` works though.

**Reply by aalmiray** [reply]
Found the bug. Setting `docker` at the packagers level does not propagate the value of `active`. Workaround is to set `docker` inside the `distribution`. Fix is on the way.

**Reply by aalmiray** [reply]
Filed https://github.com/jreleaser/jreleaser/issues/1234

**Comment by gotson** [comment]
Ok i am trying again with the env variables, and i am stumbling on another issue:

```
[INFO]  Publishing distributionss]
[INFO]    - Publishing komga distribution
[INFO]      [docker] publishing komga distribution
[INFO]      [docker] pushing to gotson/komga-docker is disabled. Skipping
[ERROR]     [docker] WARNING! Using --password via the CLI is insecure. Use --password-stdin.
[ERROR]     [docker] Error response from daemon: Get "https://registry-1.docker.io/v2/": unauthorized: incorrect username or password
[INFO]  Writing output properties to build/jreleaser/output.properties
[ERROR] JReleaser failed after 52.914 s
```

I have `externalLogin: true` for both docker registries, so i don't know why JReleaser is trying to login. The output config in the trace log shows that the external login is properly set.

[trace.log](https://github.com/jreleaser/jreleaser/files/10881356/trace.log)

**Reply by gotson** [reply]
i tried disabling `externalLogin` and explicitly setting `username` and `password` but i still have the same problem. Could that be linked to me setting a dummy value for `JRELEASER_DOCKER_DEFAULT_PASSWORD` ?

**Reply by aalmiray** [reply]
The cause is duplicate registries matching `DEFAULT`. Reported at #1232
